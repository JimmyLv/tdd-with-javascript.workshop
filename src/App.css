.App-header {
  display: flex;
  align-items: center;
  background-color: #222;
  height: 100px;
  padding: 20px;
  color: white;
}

.App-title {
  font-size: 1.5em;
}

.App-logo {
  animation: App-logo-spin infinite 20s linear;
  height: 60px;
}

@keyframes App-logo-spin {
  from {
    transform: rotate(0deg);
  }
  to {
    transform: rotate(360deg);
  }
}
