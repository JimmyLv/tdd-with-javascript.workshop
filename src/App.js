import PropTypes from 'prop-types'
import React, { Component } from 'react'
import './App.css'
import Cart from './compoments/Cart'
import logo from './logo.svg'

class App extends Component {
  static propTypes = {
    name: PropTypes.string,
  }

  static defaultProps = {
    name: 'React',
  }

  render() {
    return (
      <div className="App">
        <header className="App-header">
          <img src={logo} className="App-logo" alt="logo" />
          <h1 className="App-title">Welcome {this.props.name}!</h1>
        </header>
        <Cart />
      </div>
    )
  }
}

export default App
