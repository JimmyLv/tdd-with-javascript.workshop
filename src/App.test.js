import { shallow } from 'enzyme'
import each from 'jest-each'
import React from 'react'
import ReactDOM from 'react-dom'
import App from './App'

describe('App Component', () => {

  it('renders without crashing', () => {
    const div = document.createElement('div')
    ReactDOM.render(<App />, div)
    ReactDOM.unmountComponentAtNode(div)
  })

  it('should shallow renders without crashing', () => {
    const wrapper = shallow(<App name={'JimmyLv'} />)

    expect(wrapper.find('.App-title')).toHaveText('Welcome JimmyLv!')
    expect(wrapper).toContainReact(<code>src/App.js</code>)
  })

  it('should change the title when trigger click', () => {
    //given
    const wrapper = shallow(<App />)
    //when
    wrapper.find('button').simulate('click')
    //then
    expect(wrapper.find('h2').text()).toEqual('2')
  })
  it('should extend Array is fine', () => {
    expect([]).toBeArray()
    expect([]).toBeEmpty()
    expect([1, 2]).toBeArray()
  })
})

describe('just use jest-each to testing multiple cases', () => {
  each`
  a    | b    | expected
  ${1} | ${1} | ${2}
  ${1} | ${2} | ${3}
  ${2} | ${1} | ${3}
  ${2} | ${2} | ${4}
`.it('returns $expected when adding $a to $b', ({ a, b, expected }) => {
    expect(a + b).toBe(expected)
  })

})
