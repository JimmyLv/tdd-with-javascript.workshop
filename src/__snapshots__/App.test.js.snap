// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App Component renders correctly 1`] = `
<div
  className="App"
>
  <header
    className="App-header"
  >
    <img
      alt="logo"
      className="App-logo"
      src="logo.svg"
    />
    <h1
      className="App-title"
    >
      Welcome 
      React
      !
    </h1>
    <h2>
      1
    </h2>
  </header>
  <p
    className="App-intro"
  >
    <button
      onClick={[Function]}
    >
      +1
    </button>
    To get started, edit 
    <code>
      src/App.js
    </code>
     and save to reload.
  </p>
  <div>
    <div>
      <h1>
        React Shopping Cart
      </h1>
    </div>
    <div
      className="price"
    >
      总价：
      0
    </div>
    <div>
      <button
        className="add"
        onClick={[Function]}
      >
        添加商品
      </button>
      <div
        className="backdrop"
        onClick={[Function]}
      />
      <form
        onSubmit={[Function]}
        style={
          Object {
            "display": "none",
          }
        }
      >
        <h3>
          添加商品
        </h3>
        <input
          onChange={[Function]}
          placeholder="请输入商品编号"
          type="text"
          value=""
        />
        <button
          className="confirm"
          type="submit"
          value="Submit"
        >
          确定
        </button>
      </form>
    </div>
  </div>
</div>
`;
